sklearn/tests/test_naive_bayes.py

```diff
@@ -805,43 +809,83 @@ def test_alpha():
     prob = np.array([[1, 0], [0, 1]])
     assert_array_almost_equal(nb.predict_proba(X), prob)
 
-    nb = MultinomialNB(alpha=0.0)
+    nb = MultinomialNB(alpha=0.0, force_alpha=False)
     with pytest.warns(UserWarning, match=msg):
         nb.partial_fit(X, y, classes=[0, 1])
     with pytest.warns(UserWarning, match=msg):
         nb.fit(X, y)
     prob = np.array([[2.0 / 3, 1.0 / 3], [0, 1]])
     assert_array_almost_equal(nb.predict_proba(X), prob)
 
-    nb = CategoricalNB(alpha=0.0)
+    nb = CategoricalNB(alpha=0.0, force_alpha=False)
     with pytest.warns(UserWarning, match=msg):
         nb.fit(X, y)
     prob = np.array([[1.0, 0.0], [0.0, 1.0]])
     assert_array_almost_equal(nb.predict_proba(X), prob)
 
     # Test sparse X
     X = scipy.sparse.csr_matrix(X)
-    nb = BernoulliNB(alpha=0.0)
+    nb = BernoulliNB(alpha=0.0, force_alpha=False)
     with pytest.warns(UserWarning, match=msg):
         nb.fit(X, y)
     prob = np.array([[1, 0], [0, 1]])
     assert_array_almost_equal(nb.predict_proba(X), prob)
 
-    nb = MultinomialNB(alpha=0.0)
+    nb = MultinomialNB(alpha=0.0, force_alpha=False)
     with pytest.warns(UserWarning, match=msg):
         nb.fit(X, y)
     prob = np.array([[2.0 / 3, 1.0 / 3], [0, 1]])
     assert_array_almost_equal(nb.predict_proba(X), prob)
 
 
+def test_check_alpha():
+    """The provided value for alpha must only be
+    used if alpha < _ALPHA_MIN and force_alpha is True.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/10772
+    """
+    _ALPHA_MIN = 1e-10
+    X = np.array([[2, 1], [1, 1]])
+    y = np.array([0, 1])
+    b = BernoulliNB(alpha=0, force_alpha=True)
+    b.fit(X, y)
+    assert b._alpha == 0
+
+    alphas = np.array([0.0, 1.0])
+
+    b = BernoulliNB(alpha=alphas, force_alpha=True)
+    b.fit(X, y)
+    assert_array_equal(b._alpha, alphas)
+
+    msg = (
+        "alpha too small will result in numeric errors, setting alpha = %.1e"
+        % _ALPHA_MIN
+    )
+    b = BernoulliNB(alpha=0, force_alpha=False)
+    with pytest.warns(UserWarning, match=msg):
+        b.fit(X, y)
+        assert b._alpha == _ALPHA_MIN
```

Comment:
From memory, I see why `_check_alpha` was checked directly in a previous iteration of this PR. If the alternative is to store a `_alpha`, I prefer `_check_alpha` directly as long as there is a small comment explaining why `_check_alpha` is checked.

Storing `_alpha` only for testing purposes means the model gets slightly bigger without a runtime use case.